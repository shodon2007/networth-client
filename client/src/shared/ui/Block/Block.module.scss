.block {
	background: var(--bg-gradient-color);
	padding: 48px 80px;
	height: max-content;
	border-radius: 24px;
}
