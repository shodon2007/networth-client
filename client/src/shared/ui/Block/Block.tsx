import {FC, ReactNode} from "react";
import cls from "./Block.module.scss";

interface BlockProps {
	children: ReactNode;
}

const Block: FC<BlockProps> = ({children}) => {
	return <div className={cls.block}>{children}</div>;
};

export default Block;
