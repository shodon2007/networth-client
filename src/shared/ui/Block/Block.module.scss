.block {
	padding: 40px 24px;
	height: max-content;
	border-radius: 24px;
}

.block.default {
	background: var(--bg-gradient-color);
}

.block.brighter {
	background: var(--colorful-gradient);
}

.block * {
	color: var(--white-color);
}
