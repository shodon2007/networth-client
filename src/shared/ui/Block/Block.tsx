import {FC, ReactNode} from "react";
import cls from "./Block.module.scss";
import classNames from "src/shared/lib/classNames";

export enum BlockTypes {
	DEFAULT = "default",
	BRIGHTER = "brighter",
}
interface BlockProps {
	children: ReactNode;
	type?: BlockTypes;
	className?: string;
}

const Block: FC<BlockProps> = ({
	children,
	type = "default",
	className = "",
}) => {
	return (
		<div className={classNames(cls.block, {}, [cls[type], className])}>
			{children}
		</div>
	);
};

export default Block;
