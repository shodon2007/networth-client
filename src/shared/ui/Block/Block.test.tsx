import {render} from "@testing-library/react";
import Block from "./Block";

describe("Testing Block.tsx", () => {
	it("should be", () => {
		const component = render(
			<Block>
				<div>shodon</div>
			</Block>,
		);
		const block = component.getByTestId("block");
		expect(block).toBeInTheDocument();
		expect(block).toMatchSnapshot();
	});
});
