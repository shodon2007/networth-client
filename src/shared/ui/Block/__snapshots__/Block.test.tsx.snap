// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Block.tsx should be 1`] = `
<div
  class="block default"
  data-testid="block"
>
  <div>
    shodon
  </div>
</div>
`;
